### PR TITLE
Remove boost regex

### PR DESCRIPTION
**Pull request by OznOg**
This PR remove completely boost regex from code. Was kept for redhat workaround on fedora 32, but it is now fixed.

### CODE REVIEW

File: game/players.cc [hunk 1]
```diff
@@ -131,7 +131,7 @@ void Players::filter_internal() {
 			});
 		}
 // 		for (auto const& p: m_players) {
-// 			if (regex_search(p.name, regex(m_filter, regex_constants::icase))) filtered.push_back(p);
+// 			if (regex_search(p.name, regex(m_filter, std::regex_constants::icase))) filtered.push_back(p);
 // 		}
```
Comment (Baklap4) [inline game/players.cc:135]: Hmmm why did we comment this code out? Let's remove it if we don't intend to use player filtering.

File: game/songparser.cc [hunk 2]
```diff
@@ -145,7 +145,7 @@ void SongParser::guessFiles () {
 			file.clear();
 		}
 		if (file.empty()) { missing = true; }
-		regexps.emplace_back (p.second, regex_constants::icase);
+		regexps.emplace_back (p.second, std::regex_constants::icase);
```
Comment (Baklap4) [inline game/songparser.cc:148]: remove space between method and braces

File: CMakeLists.txt [hunk 3]
```diff
@@ -30,10 +30,6 @@ if(CMAKE_BUILD_TYPE STREQUAL "Release")
 	add_definitions("-DNDEBUG")
 endif()
 
-if (USE_BOOST_REGEX)
-        add_definitions(-DUSE_BOOST_REGEX)
-endif()
-
 option(USE_BOOST_FS "Whether to use boost filsystem instead of c++17" FALSE)
```
Comment (Baklap4) [inline CMakeLists.txt:33]: This option may be deleted too

**Review by Baklap4 — CHANGES_REQUESTED** [review]
Looks good overal. Is this also an issue in lower versions of fedora? If yes do we still want to support that or get rid of that support?

**Comment by OznOg** [comment]
> Looks good overal. Is this also an issue in lower versions of fedora? If yes do we still want to support that or get rid of that support?

fedora 31 is not supported anymore.

File: CMakeLists.txt (hunk 3 above)
Comment (OznOg) [reply CMakeLists.txt:33]: there never was such an option (forgot to add it...)

File: game/players.cc (hunk 1 above)
Comment (OznOg) [reply game/players.cc]: done

File: game/songparser.cc (hunk 2 above)
Comment (OznOg) [reply game/songparser.cc]: done

**Review by Baklap4 — APPROVED** [review]
LGTM

**Pull request merged**